be/src/exprs/array_functions.cpp

```diff
@@ -1594,4 +1594,137 @@ StatusOr<ColumnPtr> ArrayFunctions::concat(FunctionContext* ctx, const Columns&
     }
 }
 
+template <bool with_length>
+void _array_slice_item(ArrayColumn* column, size_t index, ArrayColumn* dest_column, int64_t offset, int64_t length) {
+    auto& dest_offsets = dest_column->offsets_column()->get_data();
+    if (!offset) {
+        dest_offsets.emplace_back(dest_offsets.back());
+        return;
+    }
+
+    Datum v = column->get(index);
+    const auto& items = v.get<DatumArray>();
+
+    if (offset > 0) {
+        // because offset start with 1.
+        --offset;
+    } else {
+        offset += items.size();
+    }
+
+    auto& dest_data_column = dest_column->elements_column();
+    int64_t end;
+    if constexpr (with_length) {
+        end = std::max((int64_t)0, std::min((int64_t)items.size(), (offset + length)));
+    } else {
+        end = items.size();
+    }
+    offset = (offset > 0 ? offset : 0);
```

Comment:
it seems that end should be re-computed after negative offset is adjusted to zero

Reply:
I think `offset` must not negative, because line1612 was handle negative offset, but I don't want modify any implemention in this PR, so I wasn't remove it